---
title: "gsemAnalysis"
author: "Jack Lovell"
date: "4/17/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##install packages
```{r}
library(devtools)
install_github("GenomicSEM/GenomicSEM")
require(GenomicSEM)
```

##load libraries
```{r}
library(usethis)
library(devtools)
library(tidyverse)
library(GenomicSEM)
```

## 2, explore data using major depressive disorder file
#check data size and variables
#check linear relationship between variables
#check summery and plots

##variable explanation:
#CHR: chromosome of the genetic variant
#SNP: variant ID
#BP: base-pair position of the genetic variant
#A1: effect allele
#A2: other allele
#FRQ_A_135458: Allele 1 frequency among cases (135458 is case sample size)
#FRQ_U_344901:Allele 1 frequency among controls (344901 is control sample size)
#INFO: how much learned from imputation
#OR: odds-ratio for the effect allele
#SE: standard error
```{r}
m<-read.delim('mdd')
print(as.tibble(m)) #data for major depressive disorder
lmod<-lm(INFO~OR, data=m)
summary(m)
plot(m)
plot(lm(residuals(lmod)~m$SE))
```

##Munge
#munged files: substance use disorder, post traumatic stress disorder, major depressive disorder
```{r}
setwd('~/Desktop/gsemProj/')
munge(files=c('./opi.DEPvEXP_trans.ancestry.noAF.tbl','./pts_eur_freeze2_overall.results'),
      hm3='./w_hm3.noMHC.snplist', trait.names = c("sud","ptsd"))
```

```{r}
munge(files=c('./mdd'),
      hm3='./w_hm3.noMHC.snplist', N=480359, trait.names = c("mdd"))
```


##Cov matrix
#calculate the covariance matrix using the munged files
```{r}
traits <- c("sud.sumstats.gz", "ptsd.sumstats.gz","mdd.sumstats.gz")
sample.prev <- c(.51,.15,.28)
population.prev <- c(.23,.08,.07)
ld <- "eur_w_ld_chr/"
wld <- "eur_w_ld_chr/"
trait.names<-c("SUD","PTSD","mdd")
LDSCoutput <- ldsc(traits, sample.prev, population.prev, ld, wld, trait.names)
```

#check covariance matrix output
##explanation:
#S:the covariance matrix (on the liability scale for #case/control designs).
#V:the sampling covariance matrix in the format expected by lavaan.
#I:the matrix of LDSC intercepts and cross-trait (i.e., bivariate) intercepts.
#N:contains the sample sizes (N) for the heritabilities and sqrt(N1N2) for the co-heritabilities. These are the sample sizes provided in the munging process.
#m:the number of SNPs used to construct the LD score.

```{r}
LDSCoutput
```


##check the standard errors of the ld-score regression
```{r}
k<-nrow(LDSCoutput$S)
SE<-matrix(0, k, k)
SE[lower.tri(SE,diag=TRUE)] <-sqrt(diag(LDSCoutput$V))
SE
```

##fit common factor model
#common factor model foumula: commonfactor.model<-'F1=~ NA*SUD + PTSD + mdd F1~~1*F1'
```{r}
CommonFactor_DWLS<- commonfactor(covstruc = LDSCoutput, estimation="DWLS")
CommonFactor_DWLS$results
```


```{r}
GWISmodel <- 'mdd ~ SUD + PTSD
SUD~~PTSD'
GWISoutput<-usermodel(covstruc = LDSCoutput, estimation = "DWLS", model = GWISmodel, CFIcalc = TRUE, std.lv = FALSE, imp_cov = FALSE)
GWISoutput$results
```



